tests/risks/test_effect.py

```diff
@@ -398,3 +411,98 @@
 #                                                   source=lambda index: pd.Series(0.1, index=index))
 #
 #     assert np.allclose(from_yearly(0.1, time_step)*50, em(simulation.get_population().index))
+
+
+class CustomExposureRisk(Component):
+    """Risk where we define the exposure manually."""
+
+    @property
+    def name(self) -> str:
+        return self.risk
+
+    def __init__(self, risk: str):
+        super().__init__()
+        self.risk = EntityString(risk)
+
+    # noinspection PyAttributeOutsideInit
+    def setup(self, builder: Builder):
+        builder.value.register_value_producer(
+            f"{self.risk.name}.exposure",
+            source=self.get_exposure,
+        )
+
+    def get_exposure(self, index: pd.Index) -> pd.Series:
+        data = pd.Series([0.5, 1, 1.5, 1.75, 2, 3, 4, 5, 5.5, 10], index=index)
+        return data
+
+
+def _setup_risk_simulation(
+    config: LayeredConfigTree,
+    plugins: LayeredConfigTree,
+    risk: Union[str, Risk],
+    data: Dict[str, Any],
+    has_risk_effect: bool = True,
+) -> InteractiveContext:
+    components = [TestPopulation(), risk]
+    if has_risk_effect:
+        components.append(SI("some_disease"))
```

Comment:
I don't understand why this needs to be in an iff block. I feel like components should just be declared with these components in it but I don't think this fundamentally changes your test.